aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -32,11 +32,16 @@ namespace native {
 template<typename scalar_t>
 scalar_t dot_impl(int64_t n, scalar_t *x, int64_t incx, scalar_t *y, int64_t incy);
 
-static void make_offset2bag(const Tensor &offsets, const Tensor &indices, Tensor& offset2bag) {
-  offset2bag.index_add_(
-      0, offsets, at::ones_like(offsets, LEGACY_CONTIGUOUS_MEMORY_FORMAT)); // offset2bag = [1 0 1 0 1]
-  offset2bag[0] -= 1;                     // offset2bag = [0 0 1 0 1]
-  offset2bag = offset2bag.cumsum(0);     // offset2bag = [0 0 1 1 2]
+static void make_offset2bag(const Tensor &offsets, Tensor& offset2bag) {
+  AT_DISPATCH_INDEX_TYPES(offsets.scalar_type(), "make_offset2bag", [&] () {
+    auto index_data = offsets.contiguous().data_ptr<index_t>();
```

Comment:
offsets.contiguous() is a temporary that will be deleted, you should have a local var for it